google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITCommitTimestampTest.java

```diff
@@ -0,0 +1,319 @@
+            .set("K")
+            .to("a")
+            .set("T1")
+            .to(Timestamp.MAX_VALUE)
+            .build());
+  }
+
+  @Test
+  public void invalidColumnOption() throws Exception {
+    // error_catalog error DDLStatementWithError
+    expectedException.expectCause(isSpannerException(ErrorCode.INVALID_ARGUMENT));
+    expectedException.expectMessage("Option: bogus is unknown.");
+    String statement = "ALTER TABLE T ALTER COLUMN T3 SET OPTIONS (bogus=null)";
+    dbAdminClient
+        .updateDatabaseDdl(instanceId, databaseId, ImmutableList.of(statement), null)
+        .get();
+  }
+
+  @Test
+  public void invalidColumnOptionValue() throws Exception {
+    // error_catalog error DDLStatementWithErrors
+    expectedException.expectCause(isSpannerException(ErrorCode.INVALID_ARGUMENT));
+    expectedException.expectMessage("Errors parsing Spanner DDL statement");
+    String statement = "ALTER TABLE T ALTER COLUMN T3 SET OPTIONS (allow_commit_timestamp=bogus)";
+    dbAdminClient
+        .updateDatabaseDdl(instanceId, databaseId, ImmutableList.of(statement), null)
+        .get();
+  }
+
+  @Test
+  public void invalidColumnType() throws Exception {
+    // error_catalog error OptionErrorList
+    expectedException.expect(isSpannerException(ErrorCode.FAILED_PRECONDITION));
+    expectedException.expectMessage("Option only allowed on TIMESTAMP columns");
+    String statement = "ALTER TABLE T ADD COLUMN T4 INT64 OPTIONS (allow_commit_timestamp=true)";
+    try {
+      dbAdminClient
+          .updateDatabaseDdl(instanceId, databaseId, ImmutableList.of(statement), null)
+          .get();
+    } catch (ExecutionException e) {
+      throw SpannerExceptionFactory.newSpannerException(e.getCause());
+    }
+  }
+
+  private void alterColumnOption(String databaseId, String table, String opt) throws Exception {
+    String statement =
+        "ALTER TABLE "
+            + table
+            + " ALTER COLUMN ts"
+            + " SET OPTIONS (allow_commit_timestamp="
+            + opt
+            + ")";
+    dbAdminClient
+        .updateDatabaseDdl(instanceId, databaseId, ImmutableList.of(statement), null)
+        .get();
+  }
+
+  private void writeAndVerify(DatabaseClient client, Timestamp ts) {
+    Timestamp commitTimestamp =
+        client.write(
+            Arrays.asList(
+                Mutation.newInsertOrUpdateBuilder("T1").set("ts").to(ts).build(),
+                Mutation.newInsertOrUpdateBuilder("T2").set("ts").to(ts).build(),
+                Mutation.newInsertOrUpdateBuilder("T3").set("ts").to(ts).build()));
+    if (ts == Value.COMMIT_TIMESTAMP) {
+      ts = commitTimestamp;
+    }
+    assertThat(readRow(client, "T1", Key.of(ts), "ts").getTimestamp(0)).isEqualTo(ts);
+    assertThat(readRow(client, "T2", Key.of(ts), "ts").getTimestamp(0)).isEqualTo(ts);
+    assertThat(readRow(client, "T3", Key.of(ts), "ts").getTimestamp(0)).isEqualTo(ts);
+  }
+
+  @Test
+  // 1) Write timestamps in the past
```

Comment:
I would prefer these comments in Javadoc style and placed before the `@Test` annotation, i.e.:
```java
/**
 * 1) Write timestamps in the past
 * 2) ...
*/
@Test
public void interleavedTable() throws Exception {
```

Reply:
Done @ https://github.com/googleapis/java-spanner/pull/37/commits/d920e968233662d9a16c8dd74ed94f5a2ebcb819.